lib/ukfallocbuddy/tests/test_ukfallocbuddy.c

```diff
@@ -0,0 +1,189 @@
+#include <uk/test.h>
+#include <uk/fallocbuddy.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+
+static struct uk_falloc *generate(void *fa_meta, void *start_zone, unsigned long len,
+									__paddr_t *new_start, __sz fa_meta_size, unsigned long *pages)
```

Comment:
```suggestion
static struct uk_falloc *generate(void *fa_meta, void *start_zone,
				  unsigned long len, __paddr_t *new_start,
				  __sz fa_meta_size, unsigned long *pages)
```
Configure your editor to 8 spaces per tab. Make sure that you properly align parameter lists and do not exceed 80 characters per column (please fix everywhere).

Also the function name `generate` is not really fitting, I think. Maybe something like `new_falloc` would be better.